src/Functions/FunctionsConversion.h

```diff
@@ -5561,6 +5566,38 @@ class FunctionCast final : public IFunctionBase
         }
     }
 
+    template <typename EnumType>
+    WrapperType createNativeNumberToEnumWrapper(FunctionPtr function, const DataTypePtr & from_type) const
```

Comment:
Hmm, we'd basically like to avoid this:

```sql
CREATE TABLE t0 (c0 Enum('a' = 1)) ENGINE = MergeTree() ORDER BY tuple();
INSERT INTO t0 VALUES (NULL);
```

My intuition is that a proper fix would need to be further up in the stack, ideally directly in the insert stack (can't tell where exactly). I thought about `ColumnEnum::insert` as a better place but at that point I assume the conversion from `NULL` to `0` happened already so that would already be too late.